subsquid/schema.graphql

```diff
@@ -96,10 +68,136 @@ type VestingSchedule @entity {
   from: String!
   "chain event ID"
   eventId: String!
-  "{accountId}-{assetId}"
-  scheduleId: String!
+  "Vesting schedule ID from chain"
+  scheduleId: BigInt!
   "'To' account for the vesting schedule"
   to: String!
+  "Asset ID"
+  assetId: String!
   "Vesting schedule"
   schedule: Schedule!
+  "Initial locked amount"
+  totalAmount: BigInt!
+  "True if the schedule has been fully claimed"
+  fullyClaimed: Boolean!
+}
+
+type Activity @entity {
+  id: ID!
+  "ID of the event associated with this activity"
+  eventId: String!
+  "ID of the transaction associated with this activity"
+  transactionId: String!
+  "ID of the active account"
+  accountId: String!
+  "Timestamp of the block in which this activity occurred"
+  timestamp: BigInt!
+}
+
+type Asset @entity {
+  "ID of the asset in Picasso"
+  id: ID!
+  "ID of the event that added the last price"
+  eventId: String!
+  price: BigInt!
+  historicalPrices: [HistoricalAssetPrice!] @derivedFrom(field: "asset")
+}
+
+type HistoricalAssetPrice @entity {
+  id: ID!
+  "ID of the event associated with this activity"
+  eventId: String!
+  "ID of the asset"
+  asset: Asset!
+  price: BigInt!
+  timestamp: BigInt!
+}
+
+type PicassoStakingPosition @entity {
+  id: ID!
+  eventId: String!
+  positionId: String!
+  transactionId: String!
+  poolId: String!
+  amount: BigInt!
+  owner: String!
+  startTimestamp: BigInt!
+  endTimestamp: BigInt
+}
+
+type PicassoPool @entity {
```

Comment:
(Question) What is the use case for this?

Reply:
Good question. When a pool is created we can store the asset and any other information that might not be always available, so we can use it later on. For example when a user stakes on a pool, we can get the corresponding asset, which will in turn be used to query the current price from the oracle.